### DISCUSSION TITLE
MySQL `LIKE` Operator not working as expected in PHP.

### DISCUSSION THREAD

**Discussion by Geni2** [pimcore/pimcore#14585] [Q&A / Support]
I'm trying to get data from 'MySql' where the $path value matches any value to the database using PHP but the query crash when I use LIKE with '%%' but when I use LIKE without percentage '%%' it works.

I have tried this query directly to MySQL Workbench and it works perfectly.

Does anyone have an idea why it doesn't work?

```
$sourceType = 'object';
$sourceId = 189;
$path = 'test'; //int or string value

SELECT d.targetid as id, d.sourcetype as type, CONCAT(o.o_path, o.o_key) as path
FROM dependencies d
INNER JOIN objects o ON o.o_id = d.targetid AND d.targettype= 'object'
WHERE d.sourcetype = '" . $sourceType. "' AND d.sourceid = " . $sourceId . " AND CONCAT(o.o_path, o.o_key) LIKE CONCAT('%', $path, '%');
```

I have tried in this way but also doesn't work:` LIKE '%" . $path . "%' - LIKE '%$path%'`

Only when I use: `REGEXP '".$path."' ` it works but it has problems with case-sensitive value and sometimes doesn't pull all the data. If there are 4 rows that include 'test' it gets only 3 rows for example.

**Comment by SamyMP** [comment]
I think you have to use the `quote()` method to make it work, see [this file](https://github.com/pimcore/pimcore/blob/ea5c00351ac9bb3c1956468f32611d564640ed45/bundles/AdminBundle/src/Controller/Admin/DataObject/ClassificationstoreController.php#L614) for instance

```php
$condition .= $db->quoteIdentifier($fieldname) . ' LIKE ' . $db->quote('%' . $f->value . '%');
```

Also, I think your `$path` has to  be a string if you want to concatenate it with dots `.`, or put quotes around it

**Reply by Geni2** [reply]
This is the query that get executed : 
```
SELECT d.targetid as id, d.sourcetype as type, CONCAT(o.o_path, o.o_key) as path
                    FROM dependencies d
                    INNER JOIN objects o ON o.o_id = d.targetid AND d.targettype= 'object'
                    WHERE d.sourcetype = 'object' AND d.sourceid = 189 AND CONCAT(o.o_path, o.o_key) LIKE '%test%'
```

This is how I have use it: `WHERE d.sourcetype = '" . $sourceType. "' AND d.sourceid = " . $sourceId . " AND CONCAT(o.o_path, o.o_key) LIKE ".$value."`  also like this: ` LIKE  $value`.

`$value =$db->quote('%' . $path . '%');` the value is  `$value = "'%test%'"`

Still doesn't work in this way.
